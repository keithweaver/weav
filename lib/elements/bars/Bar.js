import React from 'react';
import PropTypes from 'prop-types';
import objectAssign from 'object-assign';

const Bar = (props) => {
  const {
    percent,
    style,
    barColor,
    textStyle,
    text,
    textColor,
  } = props;

  const barWrapperStyle = {
    width: '100%',
    backgroundColor: barColor,
  };
  const barStyle = {
    width: `${percent}%`,
  };
  const textDefaultText = {
    marginTop: 0,
    marginBottom: 0,
    color: textColor,
    fontFamily: 'sans-serif',
    paddingLeft: 10,
    paddingRight: 10,
    paddingTop: 3,
    paddingBottom: 3,
  };

  return (
    <div style={barWrapperStyle}>
      <div style={objectAssign({}, barStyle, style)}>
        <p style={objectAssign({}, textDefaultText, textStyle, { color: textColor })}>
          {text}
        </p>
      </div>
    </div>
  );
};

Bar.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  style: PropTypes.object,
  percent: PropTypes.number,
  barColor: PropTypes.string,
  // eslint-disable-next-line react/forbid-prop-types
  textStyle: PropTypes.object,
  text: PropTypes.string,
  textColor: PropTypes.string,
};

Bar.defaultProps = {
  style: {},
  percent: 100,
  barColor: '#ffffff',
  textStyle: {},
  text: '',
  textColor: '#2E2E2E',
};

export default Bar;
