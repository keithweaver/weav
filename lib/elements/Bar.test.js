/* eslint-disable */
// test file
import React from 'react';
import { shallow } from 'enzyme';


import Bar from './Bar';

describe('Bar', () => {
  test('Shallow render of the Bar. Basic render, no props.', () => {
    const wrapper = shallow(
      <Bar />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
