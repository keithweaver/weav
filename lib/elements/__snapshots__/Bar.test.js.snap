// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bar Shallow render of the Bar. Basic render, no props. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Bar
    barColor="#ffffff"
    percent={100}
    style={Object {}}
    text=""
    textColor="#2E2E2E"
    textStyle={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <p
          style={
            Object {
              "color": "#2E2E2E",
              "fontFamily": "sans-serif",
              "marginBottom": 0,
              "marginTop": 0,
              "paddingBottom": 3,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 3,
            }
          }
        >
          
        </p>
      </div>,
      "style": Object {
        "backgroundColor": "#ffffff",
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p
          style={
            Object {
              "color": "#2E2E2E",
              "fontFamily": "sans-serif",
              "marginBottom": 0,
              "marginTop": 0,
              "paddingBottom": 3,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 3,
            }
          }
        >
          
        </p>,
        "style": Object {
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "style": Object {
            "color": "#2E2E2E",
            "fontFamily": "sans-serif",
            "marginBottom": 0,
            "marginTop": 0,
            "paddingBottom": 3,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 3,
          },
        },
        "ref": null,
        "rendered": "",
        "type": "p",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <p
            style={
              Object {
                "color": "#2E2E2E",
                "fontFamily": "sans-serif",
                "marginBottom": 0,
                "marginTop": 0,
                "paddingBottom": 3,
                "paddingLeft": 10,
                "paddingRight": 10,
                "paddingTop": 3,
              }
            }
          >
            
          </p>
        </div>,
        "style": Object {
          "backgroundColor": "#ffffff",
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p
            style={
              Object {
                "color": "#2E2E2E",
                "fontFamily": "sans-serif",
                "marginBottom": 0,
                "marginTop": 0,
                "paddingBottom": 3,
                "paddingLeft": 10,
                "paddingRight": 10,
                "paddingTop": 3,
              }
            }
          >
            
          </p>,
          "style": Object {
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "style": Object {
              "color": "#2E2E2E",
              "fontFamily": "sans-serif",
              "marginBottom": 0,
              "marginTop": 0,
              "paddingBottom": 3,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 3,
            },
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
